fix: always escape keys in exported objects

jasper-lyons opened the pull request:
Previously we were erroneously testing to see if a value was a valid ruby key. This returned true even when the value was not a valid ruby key e.g. 

```
'A's Fake Name': {
```

```
Ghtc_Hyuik_1[1]_160x160: {
```
I've change this behavior so that all keys are wrapped in single quotes and, if any single quotes exist in the key string they are escaped e.g.

```
'A\'s Fake Name': {
```

```
'Ghtc_Hyuik_1[1]_160x160': {
```

jasper-lyons committed:
fix: always escape keys in exported objects

Previously we were erroneously testing to see if a value was a valid ruby key. This returned true even when the value was not a valid ruby key e.g. 

```
'A's Fake Name': {
```

```
Ghtc_Hyuik_1[1]_160x160: {
```
I've change this behavior so that all keys are wrapped in single quotes and, if any single quotes exist in the key string they are escaped e.g.

```
'A\'s Fake Name': {
```

```
'Ghtc_Hyuik_1[1]_160x160': {
```

aespinoza-quoin commented:
@jasper-lyons Could you please create this PR in the main project

rbelem commented on migration/v1_to_v2/data/exporters/data_exporter.rb:
```diff
@@ -143,14 +143,8 @@ def value_to_ruby_string(value, include_blank = false)
-
   def key_to_ruby(key)
-    valid_key?(key) ? key : "'#{key}'"
+    "'#{ key.gsub(/'/, "\'") }'"
```
`"'#{ key.gsub(/'/, "\'") }'"`

I think need to add one more \
like:
`"'#{ key.gsub(/'/, "\\'") }'"`

jasper-lyons committed:
fix: escape escape character so that it's rendered

jasper-lyons committed:
Merge branch 'primeroIMS:master' into fix-record-history-export

jasper-lyons committed:
Forcing a light GC every 100 attachements imported

On some installations we can see upwards of 5GB of attachment data. When we try and load all of this into rails at once, depending on the available machine memory, we can see OOM exceptions which can bring everything down! Yay.

We're injecting a manual GC every 100 attachment loads which will drastically reduce the lifetime of the attachment objects.

jasper-lyons committed:
fix: nont string keys throwing errors

coerce all potential keys into strings before processing them

jasper-lyons committed:
fix: convert array of hashes to array of pairs

jasper-lyons committed:
Fix filename issues when exporting attachements

jasper-lyons committed:
refactor: use the procedurale approach, it's cleaner

jasper-lyons committed:
fix: use much smaller batch sizes for attachment exports

jasper-lyons committed:
fix: stop overiding default batch sizes

jasper-lyons committed:
Merge branch 'reduce-attachment-memory-consumption' into fix-record-history-export
